File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Base;

namespace ReadOnlyApp {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            // process template views
            foreach(ColumnView view in gridControl1.ViewCollection)
                view.OptionsBehavior.Editable = false;

            // process visible clone views
            foreach(ColumnView view in gridControl1.Views)
                view.OptionsBehavior.Editable = false;
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Base

Namespace ReadOnlyApp
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			' process template views
			For Each view As ColumnView In gridControl1.ViewCollection
				view.OptionsBehavior.Editable = False
			Next view

			' process visible clone views
			For Each view As ColumnView In gridControl1.Views
				view.OptionsBehavior.Editable = False
			Next view
		End Sub
	End Class
End Namespace